### DISCUSSION TITLE
Kleinere Fehler im Themes "Timeline" ....

### DISCUSSION THREAD

**Discussion by ElfenTroll** [ophian/styx#28] [General]
Hallo Ian,

ich nutze ja das Theme "Timeline" und möchte es entsprechend anpassen, leider sind mir einige Fehler aufgefallen, die gefixt werden müssten bzw. für deren Behebung ich als "Ungelernter" dringend Hilfe bräuchte. Ich liste das hier mal der Reihe nach auf:

Nr. 1)
Nach Einfügen der Navi im oberen Headerbereich funktionier der Umbruch des Blogtitels in der Headeransicht leider auch nicht. Ohne Navi hattest Du ja schon einmal gefixt.

![Ashampoo_Snap_Dienstag, 15  März 2022_16h14m46s](https://user-images.githubusercontent.com/101506660/158438460-e7d8ef07-4d44-4ad1-b676-3def151d115b.png)

Nr. 2)
Im Footer werden in der Desktopansicht alle Blöcke bzw. Plugins untereinander angezeigt. Eigentlich müssten sie ja nebeneinander angezeigt werden! 

![Ashampoo_Snap_Dienstag, 15  März 2022_16h26m45s](https://user-images.githubusercontent.com/101506660/158438917-6d106ec9-aecb-45ec-8b66-daa442e08d0d.png)

Nr. 3)
Auch die Social - Links in beiden Ansichten (Desktop und Handy) werden falsch angezeigt, sie müssten auch nebeneinander erscheinen.

![Ashampoo_Snap_Dienstag, 15  März 2022_16h16m9s](https://user-images.githubusercontent.com/101506660/158439514-483d91ce-dc9b-4619-a1eb-91da13ebd048.png)

Nr. 4)
Das ist eigentlich nicht unbedingt ein Fehler, aber empfinde ich als extrem störend! In der Artikelübersicht der Timeline werden zu den Artikeln immer "Eingangsbilder" (weiße Platzhalter) angezeigt. Diese Funktion finde ich komplett überflüssig und nervig ... Die würde ich sehr gerne komplett entfernen bzw. ausklammern, auch in der Seite für die erweiterte Artikeloptionen (ist glaube ich sogar ein Plugin). Sprich die Funktion samt der Platzhalteranzeige soll bei mir komplett rausfliegen ...

![Ashampoo_Snap_Dienstag, 15  März 2022_16h15m32s](https://user-images.githubusercontent.com/101506660/158440011-8c0f39bf-6301-4c80-8372-7f0e69c005bb.png)

Nr. 5)
Ich habe das einfache "Photoblog - Plugin" installiert. Komischerweise wird aber die Anzeigeseite in dieser "Timeline" - Version komplett ohne Sidebar und mit einem weißen Rahmen angezeigt. Ich kann den Fehler dafür auch nicht finden. Das sieht natürlich so bescheuert aus und ohne Sidebar ist es auch irgendwie unlogisch.

![Ashampoo_Snap_Dienstag, 15  März 2022_18h52m33s](https://user-images.githubusercontent.com/101506660/158441562-89b73d53-46ca-49f6-925b-5c5bac90bb71.png)

Also ich würde mich extrem freuen, wenn wir diese fünf Punkte nach und nach abarbeiten könnten, wenn es Deine Zeit erlaubt. Und nochmal Sorry das ich mit solchen Sachen hier jetzt ankomme, aber ich kann ja irgendwie auch nicht so richtig was dafür ...

Gruß Michael E.

**Comment by ophian** [comment]
zu 1:
Unter Berücksichtigung das durch den Umbruch - der so vom Author nicht vorgesehen ist, aber für das folgende hamburger "icon" wohl - möglicherweise irgendetwas anderes nicht passt, schreibe dies zu deiner user.css Datei
```css
nav .navbar-header h1 {
  white-space: break-spaces;
}
```

**Comment by ophian** [comment]
zu 2:
Du hast recht!
Fixed in https://github.com/ophian/styx/commit/97e551d29a067c37591e3189716e519704a32f9f

**Comment by ophian** [comment]
zu 3:
Genau!
Fixed in https://github.com/ophian/styx/commit/23580da820677795f4193638749b7d299a71dd5e

**Comment by ophian** [comment]
zu 4:
Das ist per design, also explizit gewollt und alles ist darauf eingerichtet. Es mag Fälle geben - so wie bei dir nun - wo man das eben nicht will. Dazu gibt es immer 2 mögliche Lösungen.
Entweder man macht sich eine eigene Kopie und gestaltet diese komplett nach den eigenen Wünschen um, das heißt in den templates und in den styles. Das erfordert aber das man komplett selbst verantwortet, was man da selbst anrichtet.
Oder man nutzt die Kraft von CSS in der eigenen `user.css` uund blendet in diesem Fall das Platzhalter icon einfach aus
```css
.timeline-heading,
.archives_summary .archive-post-thumb {
  display: none;
  visibility: hidden;
}
```
und hofft auf das Beste das nicht alles auseinanderfliegt.

**Comment by ophian** [comment]
zu 5:
Vorwort:
Es ist nicht anzunehmen das jedes Plugin mit seinen Ausgabedaten komplett Rücksicht auf jedes Theme nimmt bzw nehmen kann. Ebenso wie anzunehmen ist, dass Don als Original Theme Author auch nur je daran gedacht hätte es auf alle möglichen Plugins hin abzustimmen.

Generell:
Generell gilt das entry related Plugins ihren Content wahrscheinlich schon so platzieren dass das Grundgerüst erhalten bleibt, während andere wie dieses wohl eher auf ein eigenständiges Ausgabeformat (ohne sidebars u.a.) setzen.
Man muss wissen was man damit eigentlich will bzw **was** eigentlich beabsichtigt ist in den Möglichkeiten des Plugins. Daran orientiert sich das.
Das strukturelle und konditionale Gerüst eines themes in der `index.tpl` über die `content.tpl` zur `entries.tpl` orientiert sich an der Plugin API hook Ausgabeeinhängung bzw. an einem vorliegenden entries array.

Individuell:
Was also genau willst du mt diesem Plugin erreichen? 
Warum dies und nicht die usergallery? (Ich sage das, weil ich die usergallery für die Mediathek stark weiterentwickelt habe, so dass es auch mit der Lightbox hervorragend funktioniert, während dies Plugn hier irgendwelche javascript error Ausgaben produziert, da es nicht komplett mit dem lightbox Plugin abgecheckt wurde, nehme ich an,)

zum Border für deine user.css:
```css
.dark-skin .serendipity_event_usergallery .page_content,
.dark-skin .serendipity_event_usergallery_image_display .serendipity_gallery_entry {
  border-color: #444c56;
}
```

**Reply by ophian** [reply]
Ich sehe gerade dass das ja doch das usergallery event plugin ist. Um es mit lightbox fehlerfrei nutzen zu können, muss das Lightbox Plugin in der aktiven Liste **vor** dem usergallery Plugin stehen (gespeichert).

**Reply by ophian** [reply]
Noch einmal zur Augabe des usergallery plugins:

Oft - wie in diesem Fall - definiert ein Plugin seine Ausgaben über eine eigene Template Datei.
Und oft kann man dieses Template nehmen und es sich in sein eigenes theme hineinkopieren und somit komplett oder partiell an die eigenen Bedürfnisse anpassen, wenn man weiß was man tut.
Sobald ein solches Template zuallererst im eigenen theme gefunden wird, wird die dortige Verrsion von der Ausgabelogik (des Plugins) verwendet.
In diesem Fall sind es die `plugin_usergallery.tpl` Datei für die Struktur und das `plugin_usergallery_imagedisplay.tpl` template file für die Einzelbildanzeige.

Allerdings kann das für den unerfahrenen Anwender gewisse Schwierigkeiten bedeuten. Und so gibt es eine weitere Methode, mit der man Anpassungen im eigenen Template vornehmen kann. 
Solche Seiten sind ja ebenfalls nichts anderes als statische Seiten, auch wenn sie nicht explizit durch das Statische Seiten Plugin erzeugt werden, und unterliegen oft den gleichen Gegebenheiten.
Ihr (Seiten) Name wird über die Smarty Variable `{$staticpage_pagetitle}` ausgegeben, die man somit ebenfalls vewenden kann, um bestimmte - auf diese Seite zugeschnittene Konditionen - in sein eigenes Template (meistens in der `index.tpl`) einzufügen bzw zu erzeugen.

Du möchtest also trotzdem eine sidebar Ausgabe, auch wenn es sich um dieses Plugin handelt. Also änderst du die rechte Seitenleistenausgabe in deiner `index.tpl` Datei unterhalb des schließenden `</main>` HTML Elementes.
Und fügst am Ende (_nicht per se, aber in dieser Fall vermutlich_) bei `...'false')}`diese Kondition hier ein `...'false') OR $staticpage_pagetitle == 'Beitragsbilder'}`. Das sollte eigentlich funktionieren, wenn du die richtige Stelle erwischst.

**Comment by ophian** [comment]
Fix [Add padding to .dropdown-menu popover](https://github.com/ophian/styx/commit/90ced7371025498cdc5c45459a2df63a5af8504b)

Improve
 [Add local popper js for archive popover](https://github.com/ophian/styx/commit/0b990e5d94914ff9b9242956e82b44e53b62588b)
[Load local popper js for archive popover](https://github.com/ophian/styx/commit/dff2b0be79b0578f3f79624ef461c397ec8858fc) https://github.com/ophian/styx/commit/0b990e5d94914ff9b9242956e82b44e53b62588b
[Remove outdated bootstrap CDN examples](https://github.com/ophian/styx/commit/c432fcf36a78758a5484a22fd5271422c96b5451)

**Comment by ophian** [comment]
Ich hätte noch drei, vier Empfehlungen:

Nutze **mod_rewrite** ! Das ist unter "Konfiguration - Design und Optionen - Url-Formung" einstellbar. Dann überall wo du schon `/blog/index.php?/foo/...` permalinks in Plugin und theme configs eingestellt hast die bessere URL-Formung nutzen, also beispielsweise `/blog/pages/contact.html`.

Installiere das autoupdate event Plugin für "Ein Klick"-Upgrades.

Installiere das Spamblock Bee Plugin, konfiguriere es wie in der Dokumentation ([FAQ](https://ophian.github.io/hc/en/faq/) und im [Buch](https://ophian.github.io/book/#sub_serendipity_event_spamblock)) beschrieben, und setze es an die Spitze der Plugin Liste (gespeichert). Das ist **DAS** Plugin zur Spamabwehr.

Nutze den Core Rich Text Editor (in den Eigene Einstellungen). Inkl. erlaube **HTML Kommentare** in der Konfiguration (Generelle Einstellungen). (Möglicherweise müsste aber das timeline theme dafür noch eine Anpassung bekommen. Das Pure theme hat sie schon, zum Vergleich.)

**Reply by ophian** [reply]
Zur HTML comments option ist zu sagen, dass diese nur wirklich gut mit dem Standard Pure Theme unter besonderer Rücksichtnahme des Dark Modes funktioniert. Andere themes müssten definitiv erst um den Skript load des Rich Text (WYSIWYG) Editor erweitert werden und müssten möglicherweise noch weiter am CSS schrauben, damit die Pure CSS Erweiterungen für den Dark Mode wieder rückgängig gemacht werden.

**Comment by ElfenTroll** [comment]
Hi Ian ...Mensch Du hast Dir ja richtig Gedanken gemacht, ich bin begeistert und weiß garnicht wie ich Dir dafür danken soll.Im Augenblick bin ich noch mit Job intensiv beschäftigt, aber heute Abend und / oder Morgen werde ich das alle von Dir nach und nach umsetzen bzw. abarbeiten, und Deine Vorschläge natürlich auch alle berücksichtigen / "einbauen". Trotz der "Fehler" in "Timeline" finde ich dieses Themes immer noch echt cool und die Idee mit der "Zeitleiste" schon recht innovativ ... Was noch fehlt kann man ja, wie Du gerade zeigst, gut anpassen, wenn man Ahnung hat ... die ich leider nicht habe. Ein Vorschlag zu dem Themes wäre übrigens noch der Einbau eines Umschalters zwischen Hell und Dunkel, so wie in" Pure" ... Das finde ich echt cool!Gebe dann nach und nach Feedback ob alle Anpassungen funktioniert haben. Gruß Michael E.

**Reply by ophian** [reply]
Ich würde gerne das Bugfix release herausgeben - insofern wäre es mir lieb hier (und mt den staticpage) zu Potte zu kommen, wenns geht.

**Comment by ElfenTroll** [comment]
Hallo Ian,SORRY ... Aber ich war / bin übelst erkrankt und konnte mich daher noch nicht darum kümmern, werde das aber nun heute und morgen alles umsetzen. Michael E.

**Reply by ophian** [reply]
Gute Besserung. Gesundheit geht vor. Mach dir keinen Kopf!

**Comment by ElfenTroll** [comment]
Hi,danke für Dein Verständnis ... Ich bleibe auf jeden Fall am Ball und werde alles "testen", soweit es mir dieses WE möglich ist! Einen echt "Kleinen", wie man so sagt, hätte ich dann noch: Klickt man im Themes "Timeline" ohne etwas ins Suchfeld gesetzt zu haben auf Suche kommt ein 500er Fehler .. Nicht schlimm, aber arg unelegant! Und wenn hinter der Lupe in der Navileiste des obersten Headercointers noch das Wort Suche als Bezeichnung stehen würde wäre das Spitze ... Michael E.

**Comment by ophian** [comment]
Super. Danke. Another one of type change regression errors with PHP 8 fixed!

Ich glaube das ("Suche") passt da nicht hin und ist nicht unbedingt im Sinne des Authors. Aber du könntest es ja im template hinter das icon schreiben. Im tooltip (bei mouseover) ist es doch gesetzt und sollte m.E. reichen.

**Reply by ophian** [reply]
Ich habe das Update auf Styx 3.8.1 released. Danke nochmal fürs melden.

**Comment by ophian** [comment]
Kleine Anfrage:
Sag mal, bleibt das jetzt so auf der Blog Webseite? Oder bist du noch am "gesunden"?

**Comment by ElfenTroll** [comment]
Hi,ja, habe ich gerade gesehen .. Da macht es vielleicht Sinn noch einmal neu zu installieren, ich bin ja noch nicht sehr weit, und dann hätte ich eine saubere Version bzw. Installation. Michael E.
